actionview/lib/action_view/template/resolver.rb

```diff
@@ -143,14 +140,14 @@ def clear_cache
 
     # Normalizes the arguments and passes it on to find_templates.
     def find_all(name, prefix = nil, partial = false, details = {}, key = nil, locals = [])
-      cached(key, [name, prefix, partial], details, locals) do
-        find_templates(name, prefix, partial, details)
+      cached(key, [name, prefix, partial], details) do
+        find_templates(name, prefix, partial, details, details[:locals] || locals.sort)
```

Comment:
I won't claim to understand all the layers at play here, but passing information that's not in the cache key into a cached value's calculation makes me pretty nervous. As you're deeper in this code, can you point me to why this is safe? (Or, should we instead be shoving `locals` into `details` before the cache block, when such locals are supplied?)

Reply:
> passing information that's not in the cache key into a cached value's calculation makes me pretty nervous.

It is part of the cache key, just in a round-about way (unfortunately).  The details hash is [used to calculate the first `key`](https://github.com/rails/rails/blob/7a3afc9a69842ddf3e0edbdf078f8f3bf3c4b8f7/actionview/lib/action_view/lookup_context.rb#L171-L182).  If the locals change, the return value of `DetailsKey.get(user_details)` will change.  In this function `key` is `details_key`.